src/controllers/expenses.controller.js

```diff
@@ -0,0 +1,122 @@
+const expensesService = require('../services/expenses.service');
+const userService = require('../services/user.service');
+
+const get = async (req, res) => {
+  const { userId, from, to, categories } = req.query;
+  const expenses = await expensesService.getAllExpenses();
+
+  if (!userId && !categories && (!from || !to)) {
+    return res.send(expenses);
+  }
+
+  let filteredExpenses = expenses;
+
+  if (userId) {
+    filteredExpenses = expensesService.filterExpensesById(
+      filteredExpenses,
+      userId,
+    );
+  }
+
+  if (categories) {
+    const normalizedCategories = expensesService.normalizeCategory(categories);
+
+    filteredExpenses = expensesService.filterExpensesByCategory(
+      filteredExpenses,
+      normalizedCategories,
+    );
+  }
+
+  if (from && to) {
+    const fromDate = new Date(from);
+    const toDate = new Date(to);
+
+    filteredExpenses = expensesService.filterExpensesByDate(
+      filteredExpenses,
+      fromDate,
+      toDate,
+    );
+  }
+
+  res.send(filteredExpenses);
+};
+
+const getOne = async (req, res) => {
+  const { id } = req.params;
+
+  const expense = await expensesService.getById(id);
```

Comment:
Consider adding error handling for the async call to expensesService.getById(id). If an error occurs, it should be caught and an appropriate response should be sent to the client.